src/hooks/logging.rs

```diff
@@ -0,0 +1,45 @@
+use crate::{EvaluationContext, EvaluationDetails, EvaluationError, Value};
+
+use super::{Hook, HookContext, HookHints};
+
+/// A hook that logs the evaluation lifecycle of a flag.
+pub struct LoggingHook;
```

Comment:
@MaximFischuk please consider adding a print option to the LoggingHook struct for evaluation context; it's important to print the context in a readable JSON format if it is set [according to the spec](https://github.com/open-feature/spec/blob/main/specification/appendix-a-included-utilities.md#logging-hook)